File: src/main/java/fr/lewon/bot/runner/schedule/Delay.java
package fr.lewon.bot.runner.schedule;

import java.util.concurrent.TimeUnit;

public class Delay {

    public static final Delay ZERO = new Delay(0);

    private final long amount;
    private final TimeUnit timeUnit;

    public Delay(long amount) {
        this(amount, TimeUnit.MILLISECONDS);
    }

    public Delay(long amount, TimeUnit timeUnit) {
        this.amount = amount;
        this.timeUnit = timeUnit;
    }

    public long getDelayMillis() {
        return this.timeUnit.toMillis(this.amount);
    }
}

File: src/main/kotlin/fr/lewon/bot/runner/bot/task/Delay.kt
package fr.lewon.bot.runner.bot.task

import java.util.concurrent.TimeUnit

class Delay(private val amount: Long, private val timeUnit: TimeUnit = TimeUnit.MILLISECONDS) {

    fun getDelayMillis(): Long? {
        if (amount == NEVER.amount) {
            return null
        }
        return this.timeUnit.toMillis(this.amount)
    }

    companion object {
        val NEVER = Delay(-1)
        val ZERO = Delay(0)
    }
}
